### PR TITLE
Create blob objects and add in the commit other files than text files

### PR DESCRIPTION
**Pull request by franpilot**
This PR aims to provide a way to upload files other than text files using the GitHub API.

- `repository/create-blob!`:  Function to create blob objects.

### CODE REVIEW

File: src/clj_github/changeset.clj [hunk 1]
```diff
@@ -75,17 +75,36 @@
...
+  "Creates a blob object with the encoded content.
+  Returns a new changeset with the file under path with new encoded content and sha.
+  `content` the blob's content.
+  `encoding` the encoding used for content. `utf-8` and `base64` are supported.
+  It creates a new file if it does not exist yet."
+  [{:keys [client org repo] :as changeset} path content encoding]
+  (let [{:keys [sha]} (repository/create-blob! client org repo {:content content
+                                                                :encoding encoding})]
+    (assoc-in changeset [:changes path] {:content content
+                                         :sha sha})))
```
Comment (rfhayashi) [inline src/clj_github/changeset.clj]: Wdyt of:
- Remove this function.
- Consider at commit time that if the content is a byte array we need to create the blob.

Comment (franpilot) [reply src/clj_github/changeset.clj]: Nice, let's try it out.
By encoding the byte array in base64 and create the blob, so no change in the changeset API to add content.

File: src/clj_github/repository.clj [hunk 2]
```diff
@@ -225,3 +224,22 @@
...
+
+(defn create-blob!
+  "Creates a new blob object.
+  `content` the blob's content. It must be a byte-array when not using the `encoding` argument.
+  `encoding` the encoding used for content. `utf-8` and `base64` are supported.
+
+  For details about the parameters and response format, look at https://docs.github.com/en/rest/git/blobs#create-a-blob"
+  ([client org repo content]
+   (create-blob! client org repo (byte-array->base64 content) "base64"))
+  ([client org repo content encoding]
```
Comment (rfhayashi) [inline src/clj_github/repository.clj]: Sorry, I didn't notice this before, the pattern for the wrapper functions is to have `[client org repo params]` where `params` is the body. And you can move the base64 logic to the changeset code.

**Comment by rfhayashi** [comment]
Can you update this doc as well? https://github.com/nubank/clj-github/blob/main/src/clj_github/changeset.clj#L58

File: src/clj_github/changeset.clj [hunk 3]
```diff
@@ -80,16 +81,29 @@
...
+
+(defn- change->tree-object [changeset [path content]]
   (let [base-object {:path path
                      :mode "100644"
                      :type "blob"}]
-    (case content
-      ::deleted (assoc base-object :sha nil)
+    (condp apply [content]
+      deleted? (assoc base-object :sha nil)
+      bytes? (assoc base-object :sha (content->sha-blob! changeset content))
```
Comment (rfhayashi) [inline src/clj_github/changeset.clj:102]: TIL that `bytes?` exists :)

**Review by rfhayashi — APPROVED** [review]
Thanks for that :)

**Pull request merged**